tools/dds/py-dds-server/python.cpp

```diff
@@ -0,0 +1,202 @@
+//#include <librealsense2/rs.h>
+#include <fastdds/dds/log/Log.hpp>
+#include <fastdds/dds/domain/qos/DomainParticipantQos.hpp>
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastrtps/types/DynamicType.h>
+
+
+#ifndef BUILD_SHARED_LIBS
+INITIALIZE_EASYLOGGINGPP
+#endif
+
+
+namespace {
+std::string to_string( librealsense::dds::dds_guid const & guid )
+{
+    std::ostringstream os;
+    os << guid;
+    return os.str();
+}
+}  // namespace
+
+
+// "When calling a C++ function from Python, the GIL is always held"
+// -- since we're not being called from Python but instead are calling it,
+// we need to acquire it to not have issues with other threads...
+#define FN_FWD( CLS, FN_NAME, PY_ARGS, FN_ARGS, CODE )                                                                 \
+    #FN_NAME,                                                                                                          \
+    []( CLS & self, std::function < void PY_ARGS > callback ) {                                                        \
+        self.FN_NAME( [callback] FN_ARGS {                                                                             \
+            try {                                                                                                      \
+                py::gil_scoped_acquire gil;                                                                            \
+                CODE                                                                                                   \
+            }                                                                                                          \
+            catch( ... ) {                                                                                             \
+                std::cerr << "?!?!?!!? exception in python " #CLS "." #FN_NAME " ?!?!?!?!?" << std::endl;              \
+            }                                                                                                          \
+        } );                                                                                                           \
+    }
+
+
+// Convert FastDDS Log::Entry to EasyLogging log (see ELPP_WRITE_LOG)
+#define ISNULL(E) ((E) ? (E) : "n/a")
+#define LOG_ENTRY( ENTRY, LEVEL, ... )                                                                                 \
+    do                                                                                                                 \
+    {                                                                                                                  \
+        char const * filename = ( ENTRY ).context.filename;                                                            \
+        char const * func = ( ENTRY ).context.function;                                                                \
+        if( ! func )                                                                                                   \
```

Comment:
Style comment - prefer to check if pointer is nullptr, not using operator !

Reply:
Don't agree :)